codespell_lib/_codespell.py

```diff
@@ -28,6 +28,46 @@
 # autogenerated by setuptools_scm
 from ._version import __version__ as VERSION
 
+try:
+    import shtab
+except ImportError:
+    from . import _shtab as shtab
+
+# https://github.com/iterative/shtab/blob/5358dda86e8ea98bf801a43a24ad73cd9f820c63/examples/customcomplete.py#L11-L22
+TOML_FILE = {
+    "bash": "_shtab_greeter_compgen_toml_files",
+    "zsh": "_files -g '*.toml'",
+    "tcsh": "f:*.toml",
+}
+PREAMBLE = {
+    "bash": """
+# $1=COMP_WORDS[1]
+_shtab_greeter_compgen_toml_files() {
+  compgen -d -- $1  # recurse into subdirs
+  compgen -f -X '!*?.toml' -- $1
+}
+"""
+}
+
+
+def dict_to_choices(d={}):
+    return list(d.keys())
+
+
+def dict_to_complete(d={}):
+    def f(x):
+        return ("'" + str(x[0]) + '\\:' + x[1] + "'").replace('"', r'\"')
+
+    return {'zsh': '((' + ' '.join(map(f, d.items())) + '))'}
```

Comment:
Won't we need completion entries for bash and tcsh too?

Reply:
That is because I am only similar with zsh :smile: I think it can be a TODO for bash/tcsh user who are simliar with their shell.